File: all-submissions/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {

        int left = 0;
        int currSum =0;
        int maxSum = Integer.MIN_VALUE;
        for(int right=0;right<nums.length; right++) {
            currSum += nums[right];
            if(currSum < nums[right]) {
                currSum = nums[right];
                left = right;
            }
            maxSum = Math.max(maxSum, currSum);
        }
        return maxSum;
        // int curr_subArray = nums[0];
        // int max_subArray = nums[0];

        // for (int i =1 ;i< nums.length;i++){
        //     curr_subArray= Math.max(nums[i], curr_subArray + nums[i]);
        //     max_subArray = Math.max(curr_subArray, max_subArray);
        // }
        // return max_subArray;
    }
}

File: all-submissions/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        left = 0
        right = 1
        if not nums: return 0
        currSum = nums[0]
        maxSum = nums[0]
        while(right < len(nums)):
            currSum += nums[right]
            if currSum < nums[right]:
                currSum = nums[right]
                left = right
            maxSum = max(maxSum, currSum)
            right+=1
        
        return maxSum
